components/brave_rewards/browser/counters/rewards_counter.cc

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2019 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "base/memory/weak_ptr.h"
+#include "brave/components/brave_rewards/browser/counters/rewards_counter.h"
+#include "brave/components/brave_rewards/browser/rewards_service.h"
+#include "brave/components/brave_rewards/browser/rewards_service_factory.h"
+#include "brave/components/brave_rewards/common/pref_names.h"
+#include "chrome/browser/profiles/profile.h"
+#include "content/public/browser/browser_thread.h"
+
+using brave_rewards::RewardsService;
+using brave_rewards::RewardsServiceFactory;
+
+using namespace std::placeholders;
+
+RewardsCounter::RewardsCounter(Profile* profile)
+    : profile_(profile),
+      weak_ptr_factory_(this) {}
+
+RewardsCounter::~RewardsCounter() {
+}
+
+const char* RewardsCounter::GetPrefName() const {
+  return brave_rewards::prefs::kRewardsAutoContributeSites;
+}
+
+void RewardsCounter::Count() {
+  weak_ptr_factory_.InvalidateWeakPtrs();
```

Comment:
why are you calling this here?

Reply:
This is used in Chromium counters:

```// Count the data. Call ReportResult() when finished. Tasks that are still
  // running should be cancelled to avoid reporting old results.
  virtual void Count() = 0;```